controllers/nnf_workflow_controller.go

```diff
@@ -1547,17 +1546,27 @@ func (r *NnfWorkflowReconciler) findPersistentInstance(ctx context.Context, wf *
 	return psi, err
 }
 
-func (r *NnfWorkflowReconciler) assignWorkflowAsOwner(ctx context.Context, psi *dwsv1alpha1.PersistentStorageInstance, wf *dwsv1alpha1.Workflow) error {
-	log := r.Log.WithValues("Workflow", types.NamespacedName{Name: wf.Name, Namespace: wf.Namespace}, "persistentInstance", types.NamespacedName{Name: psi.Name, Namespace: psi.Namespace})
+type workflowStatusUpdater struct {
+	workflow       *dwsv1alpha1.Workflow
+	existingStatus dwsv1alpha1.WorkflowStatus
+}
 
-	dwsv1alpha1.AddOwnerLabels(psi, wf)
+func newWorkflowStatusUpdater(w *dwsv1alpha1.Workflow) *workflowStatusUpdater {
+	return &workflowStatusUpdater{
+		workflow:       w,
+		existingStatus: (*w.DeepCopy()).Status,
+	}
+}
 
-	if err := controllerutil.SetControllerReference(wf, psi, r.Scheme); err != nil {
-		log.Error(err, "Unable to assign workflow as owner of persistentInstance")
-		return err
+func (w *workflowStatusUpdater) close(ctx context.Context, r *NnfWorkflowReconciler) error {
+	if !reflect.DeepEqual(w.workflow.Status, w.existingStatus) {
+		err := r.Update(ctx, w.workflow)
+		if !apierrors.IsConflict(err) {
```

Comment:
If this is a conflict, then you'll return nil...will we get the retry?
Maybe this should return a result that asks for a retry.

Reply:
Seems true for all our `statusUpdater`s. Maybe address this in another PR?

(I've been wanting to jump to go 1.18 and make status updater a generic)

Reply:
If I'm right, then in this instance it can't wait for another PR.  The first conflict will stop the workflow from progressing, I think.

Reply:
I won't be finished with this PR before I stop for the day.  I'd like to have more time on Monday to continue reading through it.

Reply:
That's fine with me. I'm wrapping up for the day now.